app/controllers/concerns/gdpr_consentable.rb

```diff
@@ -0,0 +1,30 @@
+module GDPRConsentable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_consented_at, only: [:create]
+  end
+
+  def gdpr_consent_check_passed?
+    return true unless gdpr_applicable?
+    consented?
+  end
+
+  private
+
+  def gdpr_enabled?
+    ActiveRecord::Type::Boolean.new.deserialize(params[:gdpr_enabled])
+  end
+
+  def consented?
+    ActiveRecord::Type::Boolean.new.deserialize(params[:consented]) || false
+  end
+
+  def gdpr_applicable?
+    gdpr_enabled? && Country[params[:country]]&.in_eea?
+  end
+
+  def set_consented_at
+    params[:consented_at] = Time.now.in_time_zone if gdpr_applicable? && consented?
+  end
+end
```

Comment:
I'd implement this as a service instead of a module, reasons for it:
* This is only used by the create action on this controller.
* Despite using private methods, they are still included and made available from the controller, polluting the controller with 4 methods it's not supposed to use.
* It's easier to test the constent logic if it's isolated in a service.

Reply:
Also, the module knows about the `create` action, which makes the module aware of where it's being used, making the module dependent on a "controller with a create action".

Reply:
Agree, although the service I had in mind was `ManageAction`, making it GDPR aware, perhaps in combination with a new service that would create a member on request (from `ManageAction`?). To be honest, I'm seeing this as a "temporary" hack. The ideal scenario for me would be to call ManageAction with all necessary params, and ManageAction would create the action, and optionally create a member (and then publish to the action queue for other services to pick up, in this case ak-processor). Do you think this would be a reasonable approach? Or would it make more sense to create a GDPR service.

Reply:
I think the approach you mention is the right one. Have a single ManageAction service that handles everything. After we work on it, we can then asses if it makes sense to delegate the Consent part to another service, but I'd like to look at how the code looks like before making that call.